aggregator/src/main/java/io/spoud/kcc/aggregator/stream/CachedContextDataManager.java

```
@@ -53,7 +54,11 @@ public synchronized List<CachedContextData> getCachedContextData() {
                     String key = next.key;
                     ContextData contextData = next.value;
 
-                    list.add(new CachedContextData(key, contextData));
+                    try {
+                        list.add(new CachedContextData(key, contextData));
+                    } catch (Exception e) {
+                        Log.warnf("Error while creating CachedContextData for key %s: %s. This context will not be considered: %s", key, contextData, e.getMessage());
```

Since we catch all Exceptions here: Can we log the full exception on debug?

---

I am a bit unsure about doing that. The CachedContextData is cached for only 10 seconds. So if we log the whole exception, we would get a huge stacktrace in the logs every 10 seconds. What do you think?

---

I agree with @lukasgisi  we can print the stack in debug